### PR TITLE
add template development guide

### PR DESCRIPTION
**Pull request by xingao267**
https://github.com/GoogleCloudPlatform/healthcare-data-protection-suite/issues/763

### CODE REVIEW

File: docs/tfengine/component_dev.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+## New Field
+
+This should be done if the resource already exists.
+
+1. Add the field in the resource's
+    [component](../../templates/tfengine/components/resources).
+
+    *Note*: If the resource uses a Cloud Foundation Toolkit module, you may need
+    to add the field there first by sending a pull request and then wait for a
```
Comment (MartinPetkov) [inline docs/tfengine/component_dev.md:46]: Nit: "then wait**ing**"

File: docs/tfengine/debug.md [hunk 2]
```diff
@@ -1,33 +1,39 @@
-# Terraform Engine Debugging Guide
+# Terraform Engine Deployment Debugging Guide
+
+## Authentication Errors
+
+1. Use `gcloud auth application-default login` for authentication, Terraform
+    use Google Cloud Application Default Credentials (ADC).
```
Comment (MartinPetkov) [inline docs/tfengine/debug.md:6]: Nit: "uses"

**Review by MartinPetkov — APPROVED** [review]

File: docs/tfengine/component_dev.md (hunk 1 above)
Comment (xingao267) [reply docs/tfengine/component_dev.md]: Done

File: docs/tfengine/debug.md (hunk 2 above)
Comment (xingao267) [reply docs/tfengine/debug.md]: Done

**Pull request merged**